### PR TITLE
service discovery: Add operator support

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

Adds service-discovery check to the operator.

### Motivation

Allow to control service-discovery from the operator.

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Agent: v7.58.0
* Cluster Agent: N/A

### Describe your test plan

* Create `kind` cluster
* Follow instructions installations instructions https://github.com/DataDog/datadog-operator/blob/main/docs/how-to-contribute.md#testing-the-operator-for-development
* Use the following configuration
  ```yaml
  apiVersion: datadoghq.com/v2alpha1
  kind: DatadogAgent
  metadata:
    name: datadog
  spec:
    global:
      env:
        - name: DD_HOSTNAME
          value: "example-kind"
      clusterName: my-example-cluster
      credentials:
        apiSecret:
          secretName: datadog-secret
          keyName: api-key
        appSecret:
          secretName: datadog-secret
          keyName: app-key
    features:
      serviceDiscovery:
        enabled: true
    override:
      nodeAgent:
        image:
          tag: 7.59.0-linux-jmx
    ```
* Use `api key` to ddstaging
* Wait 5m and check you have entries [here](https://ddstaging.datadoghq.com/ddsql/editor?q=SELECT%20%0A%20%20%20%20a.%2A%0AFROM%20%0A%20%20%20%20discovered_service%20AS%20a%0Awhere%20a.host_name%20%3D%20%27example-kind%27%0AORDER%20BY%20%0A%20%20%20%20a.start_time&query_timeframe_end=1731222219529&query_timeframe_mode=sliding&query_timeframe_start=1730617419529)

### Checklist

- [x] PR has at least one valid label: `bug`, `enhancement`, `refactoring`, `documentation`, `tooling`, and/or `dependencies`
- [x] PR has a milestone or the `qa/skip-qa` label

### CODE REVIEW

File: docs/configuration.v2alpha1.md [hunk 1]
```diff
@@ -160,6 +160,7 @@ spec:
 | features.sbom.enabled | Enable this option to activate SBOM collection. Default: false |
 | features.sbom.host.analyzers | Analyzers to use for SBOM collection. |
 | features.sbom.host.enabled | Enable this option to activate SBOM collection. Default: false |
+| features.serviceDiscovery.enabled | Enables the service discover check. Default: false |
```
Comment (hestonhoffman) [inline docs/configuration.v2alpha1.md:163]:
```suggestion
| features.serviceDiscovery.enabled | Enables the service discovery check. Default: false |
```

**Review by hestonhoffman — APPROVED** [review]
Approved with a tiny edit

**Review by levan-m — COMMENTED** [review]
I updated PR from main and CI is failing, could you please fix it?
Test case didn't work for me, the query doesn't give me any results, is this expected?

Otherwise, change looks good. Confirmed trace-agent container and DD_SERVICE_DISCOVERY are added correctly.

File: api/datadoghq/v2alpha1/datadogagent_types.go [hunk 2]
```diff
@@ -486,6 +488,14 @@ type USMFeatureConfig struct {
 	Enabled *bool `json:"enabled,omitempty"`
 }
 
+// ServiceDiscoveryFeatureConfig configures the service discovery check feature.
+type ServiceDiscoveryFeatureConfig struct {
```
Comment (levan-m) [inline api/datadoghq/v2alpha1/datadogagent_types.go:492]: Why is this a top level feature instead of being part of USM feature?

Comment (guyarb) [reply api/datadoghq/v2alpha1/datadogagent_types.go:492]: Service discovery is not a USM feature, it is a new product

**Comment by guyarb** [comment]
> Test case didn't work for me, the query doesn't give me any results, is this expected?

What are you talking about?

> Otherwise, change looks good. Confirmed trace-agent container and DD_SERVICE_DISCOVERY are added correctly.

The change does not affect trace-agent, but system-probe and agent

**Pull request merged**